File: app/text_case.py
# Pure text-case transform shared by ASS rendering (text blocks + captions):
# apply_text_case maps "upper"/"lower" to str.upper()/str.lower(), anything else passes through.
def apply_text_case(text: str, text_case: str) -> str:
    if text_case == "upper":
        return text.upper()
    if text_case == "lower":
        return text.lower()
    return text

File: static/text-case.js
// Pure text-case helpers for TextPreset.text_case ("none" | "upper" | "lower"): apply() transforms
// a string for measurement paths; cssValue() maps to a CSS text-transform value for display paths.
// Missing/unknown values behave as "none". Consumed by preview-text.js and preview-captions.js.
window.TextCase = {
  apply(text, textCase) {
    if (textCase === "upper") return text.toUpperCase();
    if (textCase === "lower") return text.toLowerCase();
    return text;
  },
  cssValue(textCase) {
    if (textCase === "upper") return "uppercase";
    if (textCase === "lower") return "lowercase";
    return "none";
  },
};
